forc-pkg/src/pkg.rs

```diff
@@ -2586,16 +2586,30 @@ pub fn check(
     let compiled_contract_deps = HashMap::new();
 
     let mut results = vec![];
-    for &node in plan.compilation_order.iter() {
+    for (idx, &node) in plan.compilation_order.iter().enumerate() {
         let pkg = &plan.graph[node];
         let manifest = &plan.manifest_map()[&pkg.id()];
+
+        // This is necessary because `CONTRACT_ID` is a special constant that's injected into the
+        // compiler's namespace. Although we only know the contract id during building, we are
+        // inserting a dummy value here to avoid false error signals being reported in LSP.
+        // We only do this for the last node in the compilation order because previous nodes
+        // are dependencies.
+        //
+        // See this github issue for more context: https://github.com/FuelLabs/sway-vscode-plugin/issues/154
+        let contract_id_value = if idx == plan.compilation_order.len() - 1 {
+            Some("0x0000000000000000000000000000000000000000000000000000000000000000".to_string())
+        } else {
+            None
+        };
```

Comment:
Feel free to ignore since this is really just a style nit but I think this could be written as
```rust
const DUMMY_CONTRACT_ID: &str = "0x0000000000000000000000000000000000000000000000000000000000000000";
let contract_id_value = (idx == plan.compilation_order.len() - 1).then(|| DUMMY_CONTRACT_ID.to_string());
```